engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/ProcessCompiler.scala

```diff
@@ -316,7 +316,7 @@ protected trait ProcessCompilerBase {
           val joinNode = node.cast[Join].getOrElse(throw new IllegalArgumentException(s"Should be used join element in node ${nodeId.id}"))
           // TODO JOIN: use correct contexts for branches
           val contexts = joinNode.branchParameters.groupBy(_.branchId).mapValuesNow(_ => validationContext)
-          transformation.transform(contexts)
+          transformation.transform(contexts).map(_.copy(globalVariables = validationContext.globalVariables))
```

Comment:
Can you put here note that it is because of TODO in `ValidationContext`?

Reply:
done